### PR TITLE
53 textures

### LINKED ISSUE #53: Textures for 3D environment

### PR DESCRIPTION
**Pull request by thervh70**
Will close #53

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java [hunk 1]
```diff
@@ -97,20 +98,78 @@ private Group renderEntities() {
...
+			for (int i = 0; i < 100; i++) {
+				for (int j = 0; j < 3; j++) {
+					double random = Math.random();
+					final Box wallPiece = new Box(0, 1, 1);
+					wallPiece.setTranslateY(j - 3);
+					wallPiece.setTranslateZ(i);
+					wallPiece.setTranslateX(h * 3 - 1.5);
+					if (random < 0.5) {
+						wallPiece.setMaterial(Style.BRICK);
+					} else if (random > 0.5 && random < 0.75) {
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java]: In this part of the code are many magic numbers. I suggest replacing the for-loop boundaries by `Track.LENGTH` and `Track.WIDTH`, and for the other numbers you'll have to create constants.
<layout,logic

File: src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java [hunk 2]
```diff
@@ -97,20 +98,78 @@ private Group renderEntities() {
...
+
+	/**
+	 * This method is for producing track pieces.
+	 * This is a working version, however refactoring is needed.
+	 * @return Group with all the track parts
+	 */
+	private Group renderTrack() {
+		Group result = new Group();
+		for (int i = 0; i < 100; i++) {
+			for (int j = 0; j < 3; j++) {
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java]: I suggest replacing the for-loop boundaries by `Track.LENGTH` and `Track.WIDTH`.
<layout,logic

File: src/main/java/nl/tudelft/ti2206/group9/gui/PopupMenu.java [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
     /** Width of the PopUpMenu. */
     private static final double WIDTH = 260;
-    /** Height of the PopUpMenu. */
+    /** Heigth of the PopUpMenu. */
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/PopupMenu.java]: Have you just reverted a typo? :P 
<documentation

**Comment by MKerkhof** [comment]
Looks good, the only thing I can spot are some checkstyle warnings like magic numbers. Ready to merge in my opinion.

File: src/main/java/nl/tudelft/ti2206/group9/gui/Style.java [hunk 4]
```diff
@@ -27,12 +28,80 @@
...
     private static final double BUTTON_HOVER_SCALE = 1.2;
 
-	/**
-     * Private empty constructor.
+     /**
+     * Method is called once to load all textures.
+     * They are loaded into Phongmaterials.
      */
-	private Style() { }
+    public static void loadTextures() {
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/Style.java]: Method is too long. I suggest putting `"nl/tudelft/ti2206/group9/gui/" in a local variable, then you don't have to split the lines.

File: src/main/java/nl/tudelft/ti2206/group9/gui/Style.java [hunk 5]
```diff
@@ -27,12 +28,80 @@
...
+    /** PILLAR material used for pillars, stack of cracked stone bricks texture. */
+    public static final PhongMaterial PILLAR = new PhongMaterial();
+
+    /** FENCE material used for fences, mossy brick stone texture.*/
+    public static final PhongMaterial FENCE = new PhongMaterial();
+
+    /** PLAYER material used for the player */
+    public static final PhongMaterial PLAYER = new PhongMaterial();
+
+    /** Scale for button hover (in pause menu) */
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/Style.java]: *Auto-generated comment*

    First sentence should end with a period.

**Comment by mpsijm** [comment]
The logic behind your changes is good. Once you have walked through all checkstyle warnings, this is ready to be merged :)
<documentation,layout,review

File: src/main/java/nl/tudelft/ti2206/group9/gui/PopupMenu.java (hunk 3 above)
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/gui/PopupMenu.java]: Yes. Yes, I did. :P
Will revert once more.

< documentation

File: src/main/java/nl/tudelft/ti2206/group9/gui/Style.java (hunk 4 above)
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/gui/Style.java]: Fixed!

< layout

File: src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java [hunk 6]
```diff
@@ -86,31 +85,131 @@ private Group renderEntities() {
...
+	}
+
+	/**
+	 * Method to return a random material for the walls.
+	 * @return The material for the wallPiece.
+	 */
+	private static PhongMaterial chooseMaterial() {
+		double random = Math.random();
+		double alpha = 0.5;
+		double beta = 0.75;
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/ExternalTicker.java]: These local fields could be final.
<review

**Comment by MKerkhof** [comment]
No checkstyle warnings left, changes look good. Can be merged if one other team member reviews.

**Comment by mdingjan** [comment]
I can't find anymore checkstyle warnings either. Javadoc is sufficient as well and the textures are lookin' good :)
Will merge!

<review

**Comment by MathiasMeuleman** [comment]
Looks nice with all the textures :)
In my opninion, performance should be increased, but that is something for another issue.

Will merge!

<review

**Pull request merged**